version: "3.8"

services:
  db:
    image: postgres:15
    container_name: postgres_db
    environment:
      POSTGRES_USER: admin
      POSTGRES_PASSWORD: password
      POSTGRES_DB: retail_inventory_db
    volumes:
      - db_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "admin", "-d", "retail_inventory_db"]
      interval: 10s
      retries: 5
    networks:
      - app-network

  app:
    platform: "linux/amd64"
    build:
      context: .
      dockerfile: Dockerfile
      args:
        NEXT_PUBLIC_CLIENTVAR: "clientvar"
    working_dir: /app
    depends_on:
      - seed
    ports:
      - "3000:3000"
    image: t3-app
    environment:
      DATABASE_URL: postgres://${DB_USERNAME}:${DB_PASSWORD}@${DB_CONTAINER_NAME}:${DB_PORT}/${DB_NAME}
    networks:
      - app-network

  seed:
    working_dir: /app
    build:
      context: .
      dockerfile: Dockerfile.seed
    depends_on:
      db:
        condition: service_healthy
        restart: true
    image: "seed"
    environment:
      DATABASE_URL: postgres://${DB_USERNAME}:${DB_PASSWORD}@${DB_CONTAINER_NAME}:${DB_PORT}/${DB_NAME}
    networks:
      - app-network

volumes:
  db_data:

networks:
  app-network:
    driver: bridge
